exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -230,15 +452,35 @@ public void setup(HashAggregate hashAggrConfig, HashTableConfig htConfig, Fragme
       throw new IllegalArgumentException("Wrong number of workspace variables.");
     }
 
-//    this.context = context;
+    this.context = context;
     this.stats = stats;
-    this.allocator = allocator;
+    this.allocator = oContext.getAllocator();
+    this.oContext = oContext;
     this.incoming = incoming;
-//    this.schema = incoming.getSchema();
     this.outgoing = outgoing;
     this.outContainer = outContainer;
+    this.operatorId = hashAggrConfig.getOperatorId();
+
+    is2ndPhase = hashAggrConfig.getAggPhase() == AggPrelBase.OperatorPhase.PHASE_2of2;
+    isTwoPhase = hashAggrConfig.getAggPhase() != AggPrelBase.OperatorPhase.PHASE_1of1;
+    canSpill = isTwoPhase; // single phase can not spill
```

Comment:
Three booleans is 2^3 ( == 8), and this space is sparse as there are dependencies (e.g. Single phase can not spill, and has no 1st/2nd). 
  Those booleans are used only in a couple of places in the code to make a logical difference, hence it'll be an overkill to subclass based on these changes. They are used in many other places for logging messages ....